packages/venia-concept/src/components/Navigation/navigation.js

```diff
@@ -179,15 +181,17 @@ class Navigation extends PureComponent {
         }));
     };
 
+    setParentId = parentId => {
+        this.setState(() => ({ parentId }));
+    };
+
     setRootNodeId = rootNodeId => {
         this.setState(() => ({ rootNodeId }));
     };
 
     setRootNodeIdToParent = () => {
-        const { categories } = this.props;
-
-        this.setState(({ rootNodeId }) => ({
-            rootNodeId: categories[rootNodeId].parentId
+        this.setState(() => ({
+            rootNodeId: this.state.parentId
         }));
```

Comment:
When using current state to construct the next state, use the first argument to the function here.

```es6
this.setState(({ parentId }) => ({ rootNodeId: parentId }))
```